packages/frontend/src/providers/TrackingProvider.tsx

```diff
@@ -0,0 +1,217 @@
+
+    const {
+        mode,
+        version,
+        rudder: { writeKey, dataPlaneUrl },
+    } = health.data || { rudder: {} };
+
+    const appContext = useMemo(
+        () => ({
+            namespace: 'lightdash',
+            name: LIGHTDASH_APP_NAME,
+            version,
+            build: version,
+        }),
+        [version],
+    );
+
+    const getLightdashPageProperties = useCallback(
+        ({ name, category, type = PageType.PAGE }: Partial<PageData> = {}) => ({
+            name,
+            category,
+            type,
+            hostname: window.location.hostname,
+            url: null,
+            path: null,
+            referrer: null,
+            initial_referrer: null,
+            search: null,
+        }),
+        [],
+    );
+
+    const lightdashContext = useMemo(
+        () => ({
+            app: appContext,
+            page: getLightdashPageProperties(pageContext),
+        }),
+        [appContext, pageContext, getLightdashPageProperties],
+    );
+
+    useEffect(() => {
+        if (rudder) {
+            setRudderAnalytics(rudder);
+        } else if (writeKey && dataPlaneUrl) {
+            rudderSDK.load(writeKey, dataPlaneUrl);
+            rudderSDK.ready(() => {
+                setRudderAnalytics(rudderSDK);
+            });
+        }
+    }, [rudder, writeKey, dataPlaneUrl]);
+
+    const page = useCallback(
+        (rudderPageEvent: PageData): void => {
+            console.log(
+                'page',
+                rudderPageEvent.category,
+                rudderPageEvent.name,
+                getLightdashPageProperties(rudderPageEvent),
+                lightdashContext,
+            );
+            rudderAnalytics?.page(
+                rudderPageEvent.category,
+                rudderPageEvent.name,
+                getLightdashPageProperties(rudderPageEvent),
+                lightdashContext,
+            );
+        },
+        [rudderAnalytics, lightdashContext, getLightdashPageProperties],
+    );
+
+    const track = useCallback(
+        ({ name, properties = {} }: EventData): void => {
+            console.log('track', `${LIGHTDASH_APP_NAME}.${name}`, properties, {
+                ...lightdashContext,
+                section: sectionContext,
+            });
+            rudderAnalytics?.track(
+                `${LIGHTDASH_APP_NAME}.${name}`,
+                properties,
+                {
+                    ...lightdashContext,
+                    section: sectionContext,
+                },
+            );
+        },
+        [rudderAnalytics, sectionContext, lightdashContext],
+    );
+    const identify = useCallback(
+        ({ id, traits }: IdentifyData) => {
+            console.log('identify', id, traits, lightdashContext);
+            if (mode && mode !== LightdashMode.DEMO) {
+                rudderAnalytics?.identify(id, traits, lightdashContext);
+            }
+        },
+        [lightdashContext, mode, rudderAnalytics],
+    );
```

Comment:
Have these functions changed? Are the schemas for the events the same as before?

Reply:
Haven't changed business logic or properties. 
I refactored some of the functions that get the app and page context to use `useMemo` and `useCallback`.
I also changed the function props since they no longer need the page/section information.

Reply:
perfect thanks, was hard to tell without the diff